fairscale/nn/misc/checkpoint_activations.py

```diff
@@ -30,27 +30,30 @@ def checkpoint_wrapper(module: nn.Module, offload_to_cpu: bool = False) -> nn.Mo
         checkpointed_module = checkpoint_wrapper(my_module, offload_to_cpu=True)
         a, b = checkpointed_module(x, y=3, z=torch.Tensor([1]))
 
+    To understand the benefits of checkpointing and the `offload_to_cpu` flag,
+    let's divide activations into 2 types: inner activations and outer
+    activations w.r.t. the checkpointed modules. The inner ones are saved
+    by activation checkpointing, the outer ones are saved by offload_to_cpu.
+
+    In terms of GPU memory savings:
+
+        - When inner ones are large in size and outer ones are small,
+          checkpointing helps a lot, offload_to_cpu may help a little.
+        - When inner ones are small and outer ones are large,
+          checkpointing helps little, offload_to_cpu helps a lot.
+        - When both inner and outer are large, both help and the
+          benefit is additive.
+
+    The first and last most likely do not benefit from the `offload_to_cpu` flag.
```

Comment:
Looks great! A couple comments:
- at first I thought the "first and last" here referred to the bullets above, but I think you mean layers?
- reflecting on it, I think the first layer may be useful to checkpoint in some cases, shall we just say the last layer? 

Something like:
```
Note: The last layer is not likely to benefit (...)
```

Reply:
Why is first layer benefiting from cpu_offloading?

Yeah, I fixed the typo. I meant first and last layers.

Reply:
If the input is very large (e.g., images? video?), you might still want to move it to CPU since you'll need it to recompute the activations for the first layer in the backward pass

Reply:
But that input is often referred outside by the training loop, so this module can't really help since the outside reference will keep that input in GPU memory, right?

Reply:
I clarified it in my last commit.